content/crutchcorn/collections/art-of-accessibility/posts/art-of-a11y-labels/index.md

```diff
@@ -0,0 +1,869 @@
+
+```html
+<!-- DO NOT DO THIS, IT IS INACCESSIBLE -->
+<form style="display: flex; gap: 1rem;">
+	<div style="display: flex; flex-direction: column;">
+        <p style="margin: 0;">Username</p>
+        <input name="username" type="text"/>
+	</div>
+	<div style="display: flex; flex-direction: column;">
+        <p style="margin: 0;">Password</p>
+	    <input name="password" type="password"/>
+    </div>
+    <div style="display: flex; flex-direction: column; justify-content: flex-end;">
+	    <button type="submit">Login</button>
+    </div>
+</form>	
+```
+
+----
+
+<form onsubmit="event.preventDefault()" style="display: flex; gap: 1rem;">
+	<div style="display: flex; flex-direction: column;">
+        <p style="margin: 0;">Username</p>
+        <input name="username" type="text"/>
+	</div>
+	<div style="display: flex; flex-direction: column;">
+        <p style="margin: 0;">Password</p>
+	    <input name="password" type="password"/>
+    </div>
+    <div style="display: flex; flex-direction: column; justify-content: flex-end;">
+        <button type="submit">Login</button>
+    </div>
+</form>	
+
+
+---
+
+Now the fields visually _look_ like they're labelled, but we've just introduced a critical accessibility issue into our app: Assistive technologies do not indicate which label belongs to which field.
+
+After all, how would a screen reader know that adjacent `input` and `p` tags are supposed to be related? After all, how would the code automatically know how to link the following HTML:
+
+```html
+<p>Sign up for our newsletter</p>
+<input name="email" />
+<p>Email</p>
+```
+
+> You might think that the `input`'s `name` attribute provides a hint to accessibility technologies, but alas this is not the case. The `name` attribute is simply there to tell the `form` which input relates to what dataset it should track.
+
+Because of this semantic ambiguity, there are two different ways of linking elements together:
+
+- Implicit element association
+- Explicit element association
+
+Let's start with "implicit element association" and go from there.
+
+# Implicit Element Association
+
+Luckily, when dealing with `input`s, there's an easy way to link a text input to a text label: simply wrap your `input` in a `label` element:
+
+```html
+<form>
+	<label>
+        Username
+        <input name="username" type="text"/>
+	</label>
+	<label>
+        Password
+	    <input name="password" type="password"/>
+    </label>
+    <button type="submit">Login</button>
+</form>	
+```
+
+---
+
+<form onsubmit="event.preventDefault()">
+	<label>
+        Username
+        <input name="username" type="text"/>
+	</label>
+	<label>
+        Password
+	    <input name="password" type="password"/>
+    </label>
+    <button type="submit">Login</button>
+</form>	
+
+---
+
+This allows screen-readers to associate elements together and read out "Text input, username" when the user has the first text input focused.
```

Comment:
Consider mentioning labels allow users to click them to focus their respective input